### ISSUE TITLE
Rebuilds fail on Windows in fullstack projects

### ISSUE THREAD

**Issue by Hamcha** [DioxusLabs/dioxus#3262]
**Problem**

`dx serve` fails to rebuild the project on Windows specifically when the change is *not* hot-reloadable. Hot-reload works just fine. 
On my linux setup, none of this happens and everything works just fine.

Every time a new build is triggered, whether by the watcher or manually by pressing R key, a step will fail like this:

> 19:17:32 [dev] Running wasm-bindgen
19:17:33 [dev] wasm-bindgen complete in 1.879154s
19:17:33 [dev] Copying server executable to: "C:\\projects\\repro-test\\target\\dx\\repro-test\\debug\\web\\server.exe" Bu    exe: "C:\\projects\\repro-test\\target\\dioxus-server\\repro-test.exe",
    assets: AssetManifest {
        assets: {},
    },
    time_taken: 3.5184063s,
}
19:17:33 [dev] Setting builder to failed state
19:17:34 [dev] Build failed: IO(Os { code: 32, kind: Uncategorized, message: "The process cannot access the file because it is being used by another process." })

<details>
<summary>Full rebuild log with verbose logging enabled</summary>

```
19:17:28 [dev] Full rebuild: triggered manually
19:17:28 [dev] Running build command...
19:17:28 [dev] Building app...
19:17:28 [dev] Executing cargo...
19:17:28 [dev] cargo args: ["--profile", "dioxus-wasm", "--target", "wasm32-unknown-unknown", "--verbose", "--features", "web", "--bin", "repro-test"]
19:17:28 [dev] Building server...
19:17:28 [dev] Building app...
19:17:28 [dev] Executing cargo...
19:17:28 [dev] cargo args: ["--profile", "dioxus-server", "--verbose", "--features", "server", "--bin", "repro-test"]
19:17:28 [dev] cargo args: ["--profile", "dioxus-wasm", "--target", "wasm32-unknown-unknown", "--verbose", "--features", "web", "--bin", "repro-test"]
19:17:28 [dev] cargo args: ["--profile", "dioxus-server", "--verbose", "--features", "server", "--bin", "repro-test"]
19:17:30 [dev] Build completed successfully - output location: "C:\\projects\\repro-test\\target\\wasm32-unknown-unknown\\dioxus-wasm\\repro-test.wasm"
19:17:30 [dev] Collecting assets ...
19:17:30 [dev] cargo args: ["--profile", "dioxus-wasm", "--target", "wasm32-unknown-unknown", "--verbose", "--features", "web", "--bin", "repro-test"]
19:17:31 [dev] Build completed successfully - output location: "C:\\projects\\repro-test\\target\\dioxus-server\\repro-test.exe"
19:17:31 [dev] Collecting assets ...
19:17:31 [dev] cargo args: ["--profile", "dioxus-server", "--verbose", "--features", "server", "--bin", "repro-test"]
19:17:32 [dev] Assembling app bundle
19:17:32 [dev] Running wasm-bindgen
19:17:33 [dev] wasm-bindgen complete in 1.879154s
19:17:33 [dev] Copying server executable to: "C:\\projects\\repro-test\\target\\dx\\repro-test\\debug\\web\\server.exe" Bu    exe: "C:\\projects\\repro-test\\target\\dioxus-server\\repro-test.exe",
    assets: AssetManifest {
        assets: {},
    },
    time_taken: 3.5184063s,
}
19:17:33 [dev] Setting builder to failed state
19:17:34 [dev] Build failed: IO(Os { code: 32, kind: Uncategorized, message: "The process cannot access the file because it is being used by another process." })
```
</details>

I have compiled the smallest amount of code that triggers this (mostly a stripped down template): https://github.com/Hamcha/repro-test

**Steps To Reproduce**

- Create a full-stack project
- Start `dx serve`
- Make any non hot-reloadable change or just press the R key

**Expected behavior**

Just like on Linux, a rebuild should work.

**Environment:**
 - Dioxus version:  v0.6.0-alpha.5
 - Rust version:    rustc 1.83.0 (90b35a623 2024-11-26)
 - OS info:         Windows 10 22H2
 - App platform:    `fullstack`

**Questionnaire**

I'm interested in fixing this myself but don't know where to start

**Comment by DogeDark** [comment]
This should be fixed in #3239. Can you test the Git version of the CLI? 

```
cargo install --git https://github.com/DioxusLabs/dioxus dioxus-cli
```

**Comment by Hamcha** [comment]
Yup it works! Thank you very much!
